src/v/cluster/types.h

```diff
@@ -245,9 +245,9 @@ struct init_tm_tx_request
       init_tm_tx_request,
       serde::version<0>,
       serde::compat_version<0>> {
-    kafka::transactional_id tx_id;
-    std::chrono::milliseconds transaction_timeout_ms;
-    model::timeout_clock::duration timeout;
+    kafka::transactional_id tx_id{};
+    std::chrono::milliseconds transaction_timeout_ms{};
+    model::timeout_clock::duration timeout{};
```

Comment:
@andijcr 

Can you explain why this change is being made? These types are named_type<T> so they should already be default initialized to well known values. So it doesn't seem like this changes anything.

Further, the commit message offers no context for what why the change is made or how it fixes anything:

```
related to structures taking part in transaction
```